tasks/connectors/qualys_was/qualys_was.rb

```diff
@@ -230,7 +230,10 @@ def domain_detail(find_from)
       end
 
       def name(find_from)
-        "#{find_from['qid']}-#{find_from['name']}"
+        return if find_from.nil?
+
+        structured_name = [find_from['name'], find_from['qid'], find_from['id']].compact.join(' - ')
```

Comment:
Good points :D Thank you for pointing them out. I'm gonna do some testing and we can discuss in the call what could be the better solutions

Reply:
In regards to the comment 

> In the case of existing connectors, I believe this would change the finding's external_unique_id which would result in new findings being created, if that's the case, can we review how the data looks after a new cr with these changes? (i.e. are old findings still showing up? are they autoclosed?)

I've done the testing. I've generated 2 files, one with the original code, one with the updated code. 

So the result if that they indeed created new findings, however, the old ones are auto closed with status "resolved". 

In case anyone want to test it as well, these are the files
[qualys_was_originalcode.json](https://github.com/KennaSecurity/toolkit/files/13971384/qualys_was_originalcode.json)
[qualys_was_newcode.json](https://github.com/KennaSecurity/toolkit/files/13971386/qualys_was_newcode.json)